pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -126,76 +132,54 @@ public PendingAckHandleImpl(PersistentSubscription persistentSubscription) {
                 completeHandleFuture();
             }
         });
+
+        internalPinnedExecutor = persistentSubscription
+                .getTopic()
+                .getBrokerService()
+                .getInternalExecutorService()
+                .getExecutor();
     }
 
     private void initPendingAckStore() {
         if (changeToInitializingState()) {
-            synchronized (PendingAckHandleImpl.this) {
-                if (!checkIfClose()) {
-                    this.pendingAckStoreFuture =
-                            pendingAckStoreProvider.newPendingAckStore(persistentSubscription);
-                    this.pendingAckStoreFuture.thenAccept(pendingAckStore -> {
-                        pendingAckStore.replayAsync(this,
-                                ((PersistentTopic) persistentSubscription.getTopic()).getBrokerService()
-                                        .getPulsar().getTransactionReplayExecutor());
-                    }).exceptionally(e -> {
-                        acceptQueue.clear();
-                        changeToErrorState();
-                        log.error("PendingAckHandleImpl init fail! TopicName : {}, SubName: {}", topicName, subName, e);
-                        return null;
-                    });
-                }
+            if (!checkIfClose()) {
+                this.pendingAckStoreFuture =
+                        pendingAckStoreProvider.newPendingAckStore(persistentSubscription);
+                this.pendingAckStoreFuture.thenAccept(pendingAckStore -> {
+                    pendingAckStore.replayAsync(this,
+                            persistentSubscription.getTopic().getBrokerService()
+                                    .getPulsar().getTransactionReplayExecutor());
+                }).exceptionally(e -> {
+                    acceptQueue.clear();
+                    changeToErrorState();
+                    log.error("PendingAckHandleImpl init fail! TopicName : {}, SubName: {}", topicName, subName, e);
+                    return null;
+                });
             }
         }
     }
 
     private void addIndividualAcknowledgeMessageRequest(TxnID txnID,
                                                         List<MutablePair<PositionImpl, Integer>> positions,
                                                         CompletableFuture<Void> completableFuture) {
-        acceptQueue.add(() -> individualAcknowledgeMessage(txnID, positions, true).thenAccept(v ->
-                completableFuture.complete(null)).exceptionally(e -> {
+        acceptQueue.add(() -> internalIndividualAcknowledgeMessage(txnID, positions, new CompletableFuture<>())
```

Comment:
why we need to create a CompletableFuture? If `internalIndividualAcknowledgeMessage` can handle completableFuture, we don't need to create a new future